File: Semester 5/PPD/proiect/examples/simple.py
"""
Simple example demonstrating the threaded n-body simulation.
"""

from shared.utils import generate_random_bodies, generate_solar_system
from threaded.simulation import ThreadedNBodySimulation


def main():
    """Run a simple n-body simulation."""
    print("=" * 60)
    print("Simple N-Body Simulation (Threaded)")
    print("=" * 60)
    
    # Generate bodies
    print("\nGenerating bodies...")
    bodies = generate_random_bodies(n=10, 
                                   mass_range=(1e20, 1e25),
                                   position_range=(-1e6, 1e6),
                                   velocity_range=(-1e3, 1e3))
    
    print(f"Created {len(bodies)} bodies")
    
    # Create simulation
    simulation = ThreadedNBodySimulation(bodies, num_threads=4)
    
    # Run simulation
    print("\nRunning simulation...")
    simulation.run(num_steps=100, dt=1.0, verbose=True)
    
    # Print statistics
    print("\n" + "=" * 60)
    print("Simulation Statistics")
    print("=" * 60)
    stats = simulation.get_statistics()
    for key, value in stats.items():
        if key != 'center_of_mass':
            print(f"{key}: {value}")
    print(f"center_of_mass: ({stats['center_of_mass'][0]:.2e}, "
          f"{stats['center_of_mass'][1]:.2e}, {stats['center_of_mass'][2]:.2e})")


if __name__ == "__main__":
    main()

File: Semester 5/PPD/proiect/examples/simple.cpp
#include "threaded_simulation.h"
#include "utils.h"
#include <iostream>
#include <iomanip>

int main() {
    std::cout << "============================================================" << std::endl;
    std::cout << "Simple N-Body Simulation (Threaded)" << std::endl;
    std::cout << "============================================================" << std::endl;
    
    // Generate bodies
    std::cout << "\nGenerating bodies..." << std::endl;
    auto bodies = generateRandomBodies(10, 1e20, 1e25, -1e6, 1e6, -1e3, 1e3);
    
    std::cout << "Created " << bodies.size() << " bodies" << std::endl;
    
    // Create simulation
    ThreadedNBodySimulation simulation(bodies, 4);
    
    // Run simulation
    std::cout << "\nRunning simulation..." << std::endl;
    simulation.run(100, 1.0, true);
    
    // Print statistics
    std::cout << "\n============================================================" << std::endl;
    std::cout << "Simulation Statistics" << std::endl;
    std::cout << "============================================================" << std::endl;
    auto stats = simulation.getStatistics();
    std::cout << "num_bodies: " << stats.num_bodies << std::endl;
    std::cout << "num_threads: " << stats.num_threads << std::endl;
    std::cout << "time_elapsed: " << std::scientific << std::setprecision(2) 
              << stats.time_elapsed << std::endl;
    std::cout << "step_count: " << stats.step_count << std::endl;
    std::cout << "kinetic_energy: " << stats.kinetic_energy << std::endl;
    std::cout << "potential_energy: " << stats.potential_energy << std::endl;
    std::cout << "total_energy: " << stats.total_energy << std::endl;
    std::cout << "center_of_mass: (" << stats.center_of_mass[0] << ", "
              << stats.center_of_mass[1] << ", " << stats.center_of_mass[2] << ")" << std::endl;
    
    return 0;
}

